easybuild/easyblocks/r/raxml.py

```diff
@@ -0,0 +1,152 @@
+##
+# Copyright 2009-2020 Ghent University
+#
+# This file is part of EasyBuild,
+# originally created by the HPC team of Ghent University (http://ugent.be/hpc/en),
+# with support of Ghent University (http://ugent.be/hpc),
+# the Flemish Supercomputer Centre (VSC) (https://www.vscentrum.be),
+# Flemish Research Foundation (FWO) (http://www.fwo.be/en)
+# and the Department of Economy, Science and Innovation (EWI) (http://www.ewi-vlaanderen.be/en).
+#
+# https://github.com/easybuilders/easybuild
+#
+# EasyBuild is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation v2.
+#
+# EasyBuild is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with EasyBuild.  If not, see <http://www.gnu.org/licenses/>.
+##
+"""
+EasyBuild support for building and installing RAxML, implemented as an easyblock
+@author: Alex Domingo (Vrije Universiteit Brussel)
+"""
+import os
+
+from easybuild.easyblocks.generic.makecp import MakeCp
+from easybuild.framework.easyconfig import CUSTOM
+from easybuild.tools.systemtools import get_cpu_features
+
+
+class EB_RAxML(MakeCp):
+    """Support for building and installing MSM."""
+    @staticmethod
+    def extra_options(extra_vars=None):
+        """Change default values of options"""
+        extra = MakeCp.extra_options()
+        # files_to_copy is not mandatory here
+        extra['files_to_copy'][2] = CUSTOM
+        return extra
+
+    def __init__(self, *args, **kwargs):
+        """RAxML easyblock constructor, define class variables."""
+        super(EB_RAxML, self).__init__(*args, **kwargs)
+
+        # Set optmization of RAxML build for host micro-architecture
+        cpuopt_label = {
+            'sse3': 'SSE3',
+            'sse4_1': 'SSE3',
+            'sse4_2': 'SSE3',
+            'avx': 'AVX',
+            'avx1.0': 'AVX',  # on macOS, AVX is indicated with 'avx1.0' rather than 'avx'
+            'avx2': 'AVX2',
+        }
+        cpu_features = set([label for feat, label in cpuopt_label.items() if feat in get_cpu_features()])
+        self.log.debug("Enabling the following CPU optimizations for RAxML: %s", ', '.join(cpu_features))
+        cpu_features.update([None])  # add generic build
+
+        # Build features supported by RAxML, grouped in non-MPI and MPI
+        build_labels = {'nompi': [None, 'PTHREADS'], 'mpi': ['MPI', 'HYBRID']}
```

Comment:
please reformat:

```suggestion
        build_labels = {
            'nompi': [None, 'PTHREADS'],
            'mpi': ['MPI', 'HYBRID'],
        }

Reply:
fixed in https://github.com/easybuilders/easybuild-easyblocks/pull/2180/commits/f7afebdd183da07959f1fa9300501ee4923e1fad